applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_adjoint_static_solver.py

```diff
@@ -124,4 +125,8 @@ def _create_mechanical_solution_strategy(self):
         return mechanical_solution_strategy
 
     def _create_solution_scheme(self):
+        rot_dofs_solver = self.settings["rotation_dofs"].GetBool()
```

Comment:
can you please use the global setting from the solver?
Just add it to the scheme_settings, then you don't have to duplicate the setting "rotation_dofs"
=> `self.scheme_settings.AddValue("rotation_dofs",self.settings["rotation_dofs"])` should do the job